### PR TITLE
Fix error with Damage Detector

### LINKED ISSUE #2538: Fix error with Damage Detector
Following error can be caused by using the darkrp pocket on the damage detector. It could likely be triggered other ways too.

```lua
[wiremod] addons/wiremod/lua/entities/gmod_wire_damage_detector.lua:108: table index is nil
  1. LinkEnt - addons/wiremod/lua/entities/gmod_wire_damage_detector.lua:108
   2. ApplyDupeInfo - addons/wiremod/lua/entities/gmod_wire_damage_detector.lua:332
    3. PostEntityPaste - addons/wiremod/lua/entities/base_wire_entity.lua:475
     4. deserialize - gamemodes/darkrp/entities/weapons/pocket/sv_init.lua:226
      5. dropPocketItem - gamemodes/darkrp/entities/weapons/pocket/sv_init.lua:284
       6. unknown - gamemodes/darkrp/entities/weapons/pocket/shared.lua:117
```

### PR DESCRIPTION
**Pull request by Aws0mee**
fixes #2538

### CODE REVIEW

**Comment by thegrb93** [comment]
What function is calling this with nil/invalid ent? Can the check be added there?

**Comment by thegrb93** [comment]
Nevermind, just saw the trace in the issue.

File: lua/entities/gmod_wire_damage_detector.lua [hunk 1]
```diff
@@ -104,7 +104,9 @@ end
 
 function ENT:LinkEnt( ent, dontupdateoutput )
 	if self.linked_entities_lookup[ent] then return false end
-
+	
+	if not ent or not ent:IsValid() then return end
```
Comment (thegrb93) [inline lua/entities/gmod_wire_damage_detector.lua:108]: Could you make this the first line of the function?

**Review by thegrb93 — CHANGES_REQUESTED** [review]

File: lua/entities/gmod_wire_damage_detector.lua (hunk 1 above)
Comment (Aws0mee) [reply lua/entities/gmod_wire_damage_detector.lua:108]: Sure thing

**Pull request merged**